### ISSUE TITLE
VCluster 0.21 marks default StorageClass in VCluster as default StorageClass in Host Cluster

### ISSUE THREAD

**Issue by a-m-w** [loft-sh/vcluster#2295]
### What happened?

We are running VCluster 0.21 on top of EKS.
We apply the following StorageClass inside the VCluster:

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: default
  annotations:
    storageclass.kubernetes.io/is-default-class: 'true'
provisioner: ebs.csi.aws.com
parameters:
  encrypted: 'true'
  "csi.storage.k8s.io/fstype": ext4
  type: gp3
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
```

If we enable syncing of StorageClasses from VCluster to Host Cluster, the synced out StorageClass also has the `storageclass.kubernetes.io/is-default-class: 'true'` annotation.
This causes issues in the host cluster, as it might lead to more than one default StorageClass or unwanted assignments of PVCs.

This behavior was not present in VCluster 0.19.4, where the Annotation was not synced out to the Host Cluster.

### What did you expect to happen?

The specific annotation for marking a StorageClass as default should not be synced out to the Host Cluster.
See also the relevant source code here: https://github.com/loft-sh/vcluster/blob/main/pkg/controllers/resources/storageclasses/syncer.go#L32

### How can we reproduce it (as minimally and precisely as possible)?

* Install VCluster 0.21 via Helm
* Create StorageClass inside VCluster
```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: default
  annotations:
    storageclass.kubernetes.io/is-default-class: 'true'
provisioner: ebs.csi.aws.com
parameters:
  encrypted: 'true'
  "csi.storage.k8s.io/fstype": ext4
  type: gp3
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
```

* Check that the synced out StorageClass in the Host Cluster also has the `storageclass.kubernetes.io/is-default-class: 'true'` annotation.

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.31.2
Kustomize Version: v5.4.2
Server Version: v1.30.6-eks-7f9249a
```

</details>

### vcluster version

<details>

```console
$ vcluster --version
vcluster version 0.21.0
```

</details>

### VCluster Config

<details>

```
                  controlPlane:
                    advanced:
                      defaultImageRegistry: ''
                      globalMetadata:
                        annotations: {}
                      headlessService:
                        annotations: {}
                        labels: {}
                      serviceAccount:
                        annotations: {}
                        enabled: true
                        imagePullSecrets: []
                        labels: {}
                        name: ''
                      virtualScheduler:
                        enabled: false
                      workloadServiceAccount:
                        annotations: {}
                        enabled: true
                        imagePullSecrets: []
                        labels: {}
                        name: ''
                    backingStore:
                      etcd:
                        deploy:
                          enabled: true
                          headlessService:
                            enabled: true
                          service:
                            enabled: true
                          statefulSet:
                            enableServiceLinks: true
                            enabled: true
                            highAvailability:
                              replicas: 3
                            persistence:
                              addVolumeMounts: []
                              addVolumes: []
                              volumeClaim:
                                accessModes:
                                  - ReadWriteOnce
                                enabled: true
                                retentionPolicy: Retain
                                size: 15Gi
                                storageClass: ''
                              volumeClaimTemplates: []
                            resources:
                              requests:
                                cpu: 20m
                                memory: 150Mi
                            scheduling:
                              affinity:
                                podAntiAffinity:
                                  requiredDuringSchedulingIgnoredDuringExecution:
                                    - labelSelector:
                                        matchLabels:
                                          app: vcluster-etcd
                                      topologyKey: kubernetes.io/hostname
                              topologySpreadConstraints:
                                - maxSkew: 1
                                  topologyKey: kubernetes.io/hostname
                                  whenUnsatisfiable: ScheduleAnyway
                                  labelSelector:
                                    matchLabels:
                                      app: vcluster-etcd
                                    matchExpressions:
                                      - key: app
                                        operator: In
                                        values:
                                          - vcluster-etcd                                   
                    coredns:
                      deployment:
                        affinity: {}
                        nodeSelector: {}
                        replicas: 2
                        tolerations: []
                        topologySpreadConstraints:
                          - labelSelector:
                              matchLabels:
                                k8s-app: kube-dns
                            maxSkew: 1
                            topologyKey: kubernetes.io/hostname
                            whenUnsatisfiable: DoNotSchedule
                      embedded: false
                      enabled: true
                    distro:
                      k8s:
                        apiServer:
                          enabled: true
                        controllerManager:
                          enabled: true
                        enabled: true
                        resources:
                          limits:
                            cpu: 100m
                            memory: 256Mi
                          requests:
                            cpu: 40m
                            memory: 64Mi
                    ingress:
                      enabled: false
                    service:
                      enabled: true
                    serviceMonitor:
                      annotations: {}
                      enabled: true
                    statefulSet:
                      highAvailability:
                        leaseDuration: 60
                        renewDeadline: 40
                        replicas: 3
                        retryPeriod: 15
                      persistence:
                        volumeClaim:
                          enabled: auto
                      resources:
                        limits:
                          ephemeral-storage: 8Gi
                          memory: 2Gi
                        requests:
                          cpu: 200m
                          ephemeral-storage: 400Mi
                          memory: 256Mi
                      scheduling:
                        affinity:
                          podAntiAffinity:
                            requiredDuringSchedulingIgnoredDuringExecution:
                              - labelSelector:
                                  matchLabels:
                                    app: vcluster
                                topologyKey: kubernetes.io/hostname
                        topologySpreadConstraints:
                          - maxSkew: 1
                            topologyKey: kubernetes.io/hostname
                            whenUnsatisfiable: ScheduleAnyway
                            labelSelector:
                              matchLabels:
                                app: vcluster
                              matchExpressions:
                                - key: app
                                  operator: In
                                  values:
                                    - vcluster                                   
                  experimental:
                    genericSync:
                      clusterRole:
                        extraRules: []
                      role:
                        extraRules: []
                    isolatedControlPlane:
                      headless: false
                    multiNamespaceMode:
                      enabled: false
                    syncSettings:
                      disableSync: false
                      rewriteKubernetesService: false
                      setOwner: true
                      targetNamespace: ''
                  integrations:
                    externalSecrets:
                      enabled: false
                    kubeVirt:
                      enabled: false
                    metricsServer:
                      enabled: false
                  policies:
                    limitRange:
                      enabled: false
                    networkPolicy:
                      annotations: {}
                      enabled: false
                      fallbackDns: 8.8.8.8
                      labels: {}
                      outgoingConnections:
                        ipBlock:
                          cidr: 0.0.0.0/0
                          except:
                            - 100.64.0.0/10
                            - 127.0.0.0/8
                            - 10.0.0.0/8
                            - 172.16.0.0/12
                            - 192.168.0.0/16
                        platform: true
                    resourceQuota:
                      enabled: false
                  rbac:
                    clusterRole:
                      enabled: auto
                    role:
                      enabled: true
                  sync:
                    fromHost:
                      csiDrivers:
                        enabled: auto
                      csiNodes:
                        enabled: auto
                      csiStorageCapacities:
                        enabled: auto
                      events:
                        enabled: true
                      ingressClasses:
                        enabled: false
                      nodes:
                        clearImageStatus: false
                        enabled: true
                        selector:
                          all: false
                          labels: {}
                        syncBackChanges: false
                      priorityClasses:
                        enabled: false
                      runtimeClasses:
                        enabled: false
                      storageClasses:
                        enabled: auto
                      volumeSnapshotClasses:
                        enabled: false
                    toHost:
                      configMaps:
                        all: false
                        enabled: true
                      endpoints:
                        enabled: true
                      ingresses:
                        enabled: false
                      networkPolicies:
                        enabled: false
                      persistentVolumeClaims:
                        enabled: true
                      persistentVolumes:
                        enabled: true
                      podDisruptionBudgets:
                        enabled: false
                      pods:
                        enabled: true
                        enforceTolerations: []
                        rewriteHosts:
                          enabled: true
                          initContainer:
                            image: public.ecr.aws/docker/library/alpine:3.20
                            resources:
                              limits:
                                cpu: 30m
                                memory: 64Mi
                              requests:
                                cpu: 30m
                                memory: 64Mi
                        translateImage: {}
                        useSecretsForSATokens: false
                      priorityClasses:
                        enabled: false
                      secrets:
                        all: false
                        enabled: true
                      serviceAccounts:
                        enabled: true
                      services:
                        enabled: true
                      storageClasses:
                        enabled: true
                      volumeSnapshotContents:
                        enabled: false
                      volumeSnapshots:
                        enabled: false
                  telemetry:
                    enabled: false
                upstreamVersion: 0.21.0
```

</details>

**Comment by saion-chatterjee** [comment]
Is there any update on this? Is it already planned? It is also important for us to be fixed soon

**Comment by cbron** [comment]
This will be included in upcoming 0.21.2 release

**Comment by saion-chatterjee** [comment]
> This will be included in upcoming 0.21.2 release

Great thanks! I wonder why the bot has marked the issue as Closed. Can you please reopen it until it has been fixed in the new release?
